backend/src/main/java/gov/cdc/usds/simplereport/db/model/TestEvent.java

```diff
@@ -154,22 +168,12 @@ public DeviceSpecimenType getDeviceSpecimenType() {
 
   // This method is temporary and eventually, this method will be deprecated in favor of
   // getResultSet()
-  public TestResult getTestResult() {
+  public TestResult getCovidTestResult() {
     final String COVID_LOINC = "96741-4";
-    if (this.results != null) {
-      Optional<Result> resultObject =
-          this.results.stream()
-              .filter(result -> COVID_LOINC.equals(result.getDisease().getLoinc()))
-              .findFirst();
-      if (resultObject.isPresent()) {
-        return resultObject.get().getTestResult();
-      }
-    }
-    return super.getResult();
-  }
-
-  @Override
-  public TestResult getResult() {
-    return getTestResult();
+    Optional<Result> resultObject =
+        this.results.stream()
+            .filter(result -> COVID_LOINC.equals(result.getDisease().getLoinc()))
+            .findFirst();
+    return resultObject.map(Result::getTestResult).orElse(null);
```

Comment:
Hmmm, I don't love the possibility of returning `null` on this method...I think we could either return `Optional<TestResult>`, or just throw an exception if someone tries to call this method on a TestEvent that doesn't have a covid result. But nulls tend to bubble up in sneaky ways.